docs/adr/0001-why-pravic.md

```diff
@@ -0,0 +1,106 @@
+
+A resource definition should be the authoritative view of the state or status of a resource.  The definition provides a configuration as code view of the desired state of the resource.  This differs from Ansible's existing task-based approach to cloud automation by abstracting the individual steps necessary to achieve the desired state, which can involve many modules and tasks which must be executed in a strict order for many cloud services today.
+```yaml
+---
+resources:
+  instance_01:
+    - name: instance_01
+      Type: AWS::EC2::Instance
+      Properties:
+        AvailabilityZone: us-east-1
+        ImageId: String
+        InstanceType: m1.medium
+        KeyName: example_key
+        SecurityGroupIds:
+          - String
+        SubnetId: resource:group_01.Properties.GroupId
+        Tags:
+          - Bar: Baz
+  elasticache_01:
+    - name: cache_01
+      Type: AWS::ElastiCache::CacheCluster
+      Properties:
+        CacheNodeType: cache.m1.medium
+        ClusterName: cache_01
+        Engine: redis
+        Tags:
+          - test: tag
+        VpcSecurityGroupIds:
+          - resource:group_01.Properties.GroupName
+  group_01:
+    - name: group_01
+      Type: AWS::EC2::SecurityGroup
+      Properties:
+        GroupDescription: some_group
+        GroupName: group_01
+        SecurityGroupEgress:
+          - {{ Egress }}
+        SecurityGroupIngress:
+          - {{ Ingress }}
+        Tags:
+          - Foo: Bar
+        VpcId: resource:vpc_01.Properties.VpcId
+  vpc_1:
+    - name: vpc_01
+      Type: AWS::EC2::VPC
+      Properties:
+        CidrBlock: 10.0.0.0/16
+```
+Generally we want to take advantage of the provider's schema as much as possible, such as the above example which is based directly on AWS' Cloud Control / CloudFormation resource schemas.  However to integrate this with Ansible it will be necessary to put some structure in place around top-level key words and the format in which resources may be accessed and referenced to ensure a consistent user experience across different Ansible components.
+
+### Dependency Resolution
+
+Cloud resources commonly depend on other resources to exist or be in a specific state before they can be created.  This often causes frustration for Ansible users as they must maintain a high degree of awareness of these relationships and ensure that tasks have the appropriate checks for resource readiness and availability.
+
+Where a vendor provides a reference schema for their resources, such as in the case of the AWS Cloud Control API, we can programmatically identify these dependencies and make decisions about the order of resource actions.
+
+In the above example, the VPC must be created before any other resources and the security group must be created before the instance and the elastic cache.  We could identify this both through the resource references in the resource definition and by analyzing the AWS schemas for these resources.
+
+### Asynchronous Execution
```

Comment:
I feel like this section could be a bit clearer. It switches back and forth between talking about Ansible's async execution and pravic's async execution. They're both useful, but the main selling point with pravic is that the playbook author doesn't need to think about it. Asynchronous execution isn't a playbook level feature, it's just built directly into the collection.